scipy/optimize/minpack.py

```diff
@@ -493,8 +498,23 @@ def curve_fit(f, xdata, ydata, p0=None, sigma=None, absolute_sigma=False,
         If True, check that the input arrays do not contain nans of infs,
         and raise a ValueError if they do. Setting this parameter to
         False may silently produce nonsensical results if the input arrays
-        do contain nans.
-        Default is True.
+        do contain nans. Default is True.
+    bounds : 2-tuple of array_like, optional
+        Lower and upper bounds on independent variables. Default is None
+        (no bounds). If provided, each element of the tuple must be either an
```

Comment:
By consistency with `least_squares` wouldn't it be better to set `bounds=(-np.inf, np.inf)` by default? That would also avoid explicitly setting it to the default value (below l. 641-642). DRY.